packages-exp/auth-exp/src/core/auth/auth_impl.ts

```diff
@@ -120,3 +203,18 @@ export function initializeAuth(
 
   return new AuthImpl(app.name, config, hierarchy);
 }
+
+/** Helper class to wrap subscriber logic */
+class Subscription<T> {
```

Comment:
should this belong in util?

Reply:
I don't think so-- this is the only place it'll be used